Submitted By: Bruce Dubbs <bdubbs@linuxfromscratch.org>
Date: 2004-08-05
Initial Package Version: 1.2.6
Origin: http://blfs-bugs.linuxfromscratch.org/show_bug.cgi?id=644
Description: libpng is not linked against libz and libm. This causes
             problems for packages that try to link against libpng
             without including "-lz -lm" in the linker flags.

$LastChangedBy$
$Date$

--- libpng-1.2.8/scripts/libpng.pc.in.orig	2005-01-04 14:06:54.182100728 +0000
+++ libpng-1.2.8/scripts/libpng.pc.in	2005-01-04 14:07:19.108311368 +0000
@@ -7,5 +7,5 @@
 Name: libpng12
 Description: Loads and saves PNG files
 Version: 1.2.8
-Libs: -L${libdir} -lpng12 -lz -lm
+Libs: -L${libdir} -lpng12
 Cflags: -I${includedir}/libpng12
--- libpng-1.2.8/scripts/makefile.linux.orig	2005-01-04 14:07:37.875458328 +0000
+++ libpng-1.2.8/scripts/makefile.linux	2005-01-04 14:11:03.918135080 +0000
@@ -34,8 +34,9 @@
 CFLAGS=-I$(ZLIBINC) -Wall -O3 -funroll-loops \
 	$(ALIGN) # $(WARNMORE) -g -DPNG_DEBUG=5
 
-LDFLAGS=-L. -Wl,-rpath,. -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) -lpng12 -lz -lm
-LDFLAGS_A=-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) libpng.a -lz -lm
+LIBS=-lz -lm
+LDFLAGS=-L. -Wl,-rpath,. -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) -lpng12
+LDFLAGS_A=-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) libpng.a $(LIBS)
 
 RANLIB=ranlib
 #RANLIB=echo
@@ -86,7 +87,7 @@
 	echo I_opts=\"-I$(INCPATH)/$(LIBNAME)\"; \
 	echo L_opts=\"-L$(LIBPATH)\"; \
 	echo R_opts=\"-Wl,-rpath,$(LIBPATH)\"; \
-	echo libs=\"-lpng12 -lz -lm\"; \
+	echo libs=\"-lpng12\"; \
 	cat scripts/libpng-config-body.in ) > libpng-config
 	chmod +x libpng-config
 
@@ -99,12 +100,12 @@
 $(LIBNAME).so.$(PNGVER): $(OBJSDLL)
 	$(CC) -shared -Wl,-soname,$(LIBNAME).so.$(PNGMAJ) \
 	-o $(LIBNAME).so.$(PNGVER) \
-	$(OBJSDLL)
+	$(OBJSDLL) $(LIBS)
 
 libpng.so.3.$(PNGMIN): $(OBJSDLL)
 	$(CC) -shared -Wl,-soname,libpng.so.3 \
 	-o libpng.so.3.$(PNGMIN) \
-	$(OBJSDLL)
+	$(OBJSDLL) $(LIBS)
 
 pngtest: pngtest.o $(LIBNAME).so
 	$(CC) -o pngtest $(CFLAGS) pngtest.o $(LDFLAGS)
